src/components/view/Details.js

```diff
@@ -10,10 +11,54 @@ var Details = React.createClass({
     return '';
   },
 
-  'fetchSubtasks': function (task) {
+  'onClickStats': function () {
+
+  },
+
+  'onClickSettings': function () {
+
+  },
+
+  'onClickHelp': function () {
+
+  },
+
+  'completeMainTask': function () {
+
+    var self = this;
+    background.updateTask(self.props.task, {'completed':!self.props.task.completed}).always(function () {
+      self.props.onUpdateTask();
+      self.state.subTasks.map(function (subtask) {
+        self.toggleSubtask(subtask, true);
+      });
+    });
+  },
+
+  'updateTaskTitle': function (newTitle) {
 
     var self = this;
-    background.fetchSubtasks(task.id).always(function (subTasks) {
+    background.updateTask(self.props.task, {'title':newTitle}).always(function () {
+      self.props.onUpdateTask();
+    });
+  },
+
+  'updateSubtaskTitle': function (subtask, newTitle) {
+
+    var self = this;
+    background.updateSubtask(subtask, {'title':newTitle}).always(self.fetchSubtasks.bind(self));
```

Comment:
don't need to bind here (self.fetchSubtasks, self) will call the fn with the correct context without needing to create a new bound fn